### ISSUE TITLE
Not seeing any output after running plotly() with julia 1.9

### ISSUE THREAD

**Issue by bryaan** [JuliaLang/IJulia.jl#1073]
Jupyter isn't outputting anything in the cells with the julia 1.9-rc3 kernel. Please fix.

**Comment by bryaan** [comment]
Upon further review this is happening after running plotly() as in 

```
using Plots
plotly()
plot(x, y)
```

Filing an issue with Plots.jl

**Comment by ajw-free** [comment]
I have the same problem with the Julia 1.9 release.  
My platform is Ubuntu Mate 23.04 and Firefox browser.

If I load Jupyter notebook, then use Plots with plotly() backend, the code below does not display the plot object. 
using Plots
plotly()
display(plot(rand(10)))

If I use the gr() backend, the plot image does display in the noteboook.

I should add that if I use Plots with plotly() from the command line REPL, it does work - a tab on my browser opens and displays the plot.

Please fix.

**Comment by ajw-free** [comment]
I compared the html code created by Jupyter Notebook (which displays nothing) with the html created from the REPL (which display properly in a browser tab).

To do this, I used savefig() to save to an html file which can be viewed in an editor.

There are differences in the first few lines which suggest that some header information is not being written.

(For gr() the files are the  same, as expected.)

To see the differences:

(1) Run this following code via the REPL:
using Plots
plotly()
p=plot(1:10)
savefig(p, "plot_from_REPL.html")
display(p)  # works and show plot in a browser tab.

(2) Run this code via Jupyter Notebook:
using Plots
plotly()
p=plot(1:10)
savefig(p, "plot_from_jupyter.html")
display(p)  # shows no display.

RESULTS 

(1) File "plot_from_REPL.html" starts with:

"<script src="https://cdn.plot.ly/plotly-2.6.3.min.js"></script>   
<div id="30958384-5167-44fe-a20a-402d93f86b5d" style="width:800px;height:300px;"></div>
etc"

(2) File "plot_from_jupyter.html" starts with:

"<div id="4708b2af-0fcd-48d0-8e3b-d902f2ca4bb6" style="width:600px;height:400px;"></div>
etc."

Clearly something is missing at the start of the html created from within the notebook.

I hope that someone can soon be allocated to fix this show-stopping bug!

**Comment by ajw-free** [comment]
As a temporary solution, one can install and use the plotlyjs() backend instead of plotly().  
It works with Julia 1.9.0 and Jupyter Notebook, and allows interactive plots.

using Pkg
Pkg.add("PlotlyJS")

From Jupyter notebook:
using Plots
plotlyjs()
plot(rand(10))

**Comment by a15355447898a** [comment]
> As a temporary solution, one can install and use the plotlyjs() backend instead of plotly(). It works with Julia 1.9.0 and Jupyter Notebook, and allows interactive plots.
> 
> using Pkg Pkg.add("PlotlyJS")
> 
> From Jupyter notebook: using Plots plotlyjs() plot(rand(10))

I tried to run this, and then, my jupyter notebook kept running without any return value
```julia
using Plots
plotlyjs()
plot(rand(10))
```

**Comment by stevengj** [comment]
`IJulia.display` is the same as `Base.display`, which is the same as what is called when the cell is evaluated.  So it should make no difference.  Indeed, your example works just fine for me without any explicit `display` call in Julia 1.9 on Mac (M4):
<img width="618" alt="image" src="https://github.com/JuliaLang/IJulia.jl/assets/2913679/6559605e-ad1b-4cce-8f8c-19fd9c2c19ac">

**Comment by stevengj** [comment]
If there is a problem with plotly, maybe report the error there?  It's not clear to me that the problem is with IJulia.

**Comment by stevengj** [comment]
It looks like the problem with Plotly is that its output MIME type `"application/vnd.plotly.v1+json"` is not registered as text, so it is getting base-64 encoded even though it should be sent as JSON text.  This is causing an error on the Python side when it tries to decode the JSON (it aborts with an error because the base64 stuff is not valid JSON).

I can fix it by defining:
```jl
Base.istextmime(::MIME"application/vnd.plotly.v1+json") = true
```
which then gives:
<img width="765" alt="image" src="https://github.com/JuliaLang/IJulia.jl/assets/2913679/9425c517-9109-461b-9c34-39c90b472608">

but it really seems like this should be defined in Plotly.jl.   (This `istextmime` method must have been defined at one point for it to work, no?  Was it removed recently?)

**Comment by a15355447898a** [comment]
> It looks like the problem with Plotly is that its output MIME type `"application/vnd.plotly.v1+json"` is not registered as text, so it is getting base-64 encoded even though it should be sent as JSON text. This is causing an error on the Python side when it tries to decode the JSON (it aborts with an error because the base64 stuff is not valid JSON).
> 
> I can fix it by defining:
> 
> ```julia
> Base.istextmime(::MIME"application/vnd.plotly.v1+json") = true
> ```
> 
> which then gives: <img alt="image" width="765" src="https://user-images.githubusercontent.com/2913679/242947596-9425c517-9109-461b-9c34-39c90b472608.png">
> 
> but it really seems like this should be defined in Plotly.jl. (This `istextmime` method must have been defined at one point for it to work, no? Was it removed recently?)

The sad thing is that I still have the problem here, which happens on top of vscode's jupyter

* This string of code keeps running without giving me a return value

<img width="966" alt="image" src="https://github.com/JuliaLang/IJulia.jl/assets/62456287/7c506532-1133-48d2-8894-f2b1ad1d04d4">

```julia
using Plots
plotlyjs()
plot(rand(10))
```

---

* This string of code runs and returns error

<img width="969" alt="image" src="https://github.com/JuliaLang/IJulia.jl/assets/62456287/0018c833-af21-4919-a076-e9d34b2083d8">

```julia
using Plots
Plots.gr()
Plots.plot(Plots.fakedata(50, 5), w=3)
```

```
┌ Error: Some Julia code in the VS Code extension crashed
└ @ VSCodeServer [c:\Users\](file:///C:/Users/)孙文彬\.vscode\extensions\julialang.language-julia-1.47.2\scripts\error_handler.jl:15
ERROR: UndefVarError: `Base64EncodePipe` not defined
Stacktrace:
  [1] send_stream(name::String, send_callback::typeof(VSCodeServer.io_send_callback))
    @ VSCodeServer.IJuliaCore [c:\Users\](file:///C:/Users/)孙文彬\.vscode\extensions\julialang.language-julia-1.47.2\scripts\packages\IJuliaCore\src\stdio.jl:152
  [2] flush(io::VSCodeServer.IJuliaCore.IJuliaStdio{Base.PipeEndpoint, typeof(VSCodeServer.io_send_callback)})
    @ VSCodeServer.IJuliaCore [c:\Users\](file:///C:/Users/)孙文彬\.vscode\extensions\julialang.language-julia-1.47.2\scripts\packages\IJuliaCore\src\stdio.jl:272
  [3] flush_all()
    @ VSCodeServer.IJuliaCore [c:\Users\](file:///C:/Users/)孙文彬\.vscode\extensions\julialang.language-julia-1.47.2\scripts\packages\IJuliaCore\src\stdio.jl:255
  [4] (::VSCodeServer.var"#198#199"{VSCodeServer.NotebookRunCellArguments, String})()
    @ VSCodeServer [c:\Users\](file:///C:/Users/)孙文彬\.vscode\extensions\julialang.language-julia-1.47.2\scripts\packages\VSCodeServer\src\serve_notebook.jl:65
  [5] withpath(f::VSCodeServer.var"#198#199"{VSCodeServer.NotebookRunCellArguments, String}, path::String)
    @ VSCodeServer [c:\Users\](file:///C:/Users/)孙文彬\.vscode\extensions\julialang.language-julia-1.47.2\scripts\packages\VSCodeServer\src\repl.jl:249
  [6] notebook_runcell_request(conn::VSCodeServer.JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint, Base.PipeEndpoint}, params::VSCodeServer.NotebookRunCellArguments)
    @ VSCodeServer [c:\Users\](file:///C:/Users/)孙文彬\.vscode\extensions\julialang.language-julia-1.47.2\scripts\packages\VSCodeServer\src\serve_notebook.jl:13
  [7] dispatch_msg(x::VSCodeServer.JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint, Base.PipeEndpoint}, dispatcher::VSCodeServer.JSONRPC.MsgDispatcher, msg::Dict{String, Any})
    @ VSCodeServer.JSONRPC [c:\Users\](file:///C:/Users/)孙文彬\.vscode\extensions\julialang.language-julia-1.47.2\scripts\packages\JSONRPC\src\typed.jl:67
  [8] serve_notebook(pipename::String, outputchannel_logger::Base.CoreLogging.SimpleLogger; crashreporting_pipename::String)
    @ VSCodeServer [c:\Users\](file:///C:/Users/)孙文彬\.vscode\extensions\julialang.language-julia-1.47.2\scripts\packages\VSCodeServer\src\serve_notebook.jl:139
  [9] top-level scope
    @ [c:\Users\](file:///C:/Users/)孙文彬\.vscode\extensions\julialang.language-julia-1.47.2\scripts\notebook\notebook.jl:32
 [10] include(mod::Module, _path::String)
    @ Base [.\](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/%E5%AD%99%E6%96%87%E5%BD%AC/Downloads/%E4%BD%BF%E7%94%A8julia%E5%92%8Cpython%E9%87%8D%E6%96%B0%E8%AE%A1%E7%AE%97%E5%A4%AA%E9%98%B3%E9%AB%98%E5%BA%A6%E8%A7%92%E9%97%AE%E9%A2%98/)Base.jl:457
 [11] exec_options(opts::Base.JLOptions)
    @ Base [.\](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/%E5%AD%99%E6%96%87%E5%BD%AC/Downloads/%E4%BD%BF%E7%94%A8julia%E5%92%8Cpython%E9%87%8D%E6%96%B0%E8%AE%A1%E7%AE%97%E5%A4%AA%E9%98%B3%E9%AB%98%E5%BA%A6%E8%A7%92%E9%97%AE%E9%A2%98/)client.jl:307
 [12] _start()
    @ Base [.\](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/%E5%AD%99%E6%96%87%E5%BD%AC/Downloads/%E4%BD%BF%E7%94%A8julia%E5%92%8Cpython%E9%87%8D%E6%96%B0%E8%AE%A1%E7%AE%97%E5%A4%AA%E9%98%B3%E9%AB%98%E5%BA%A6%E8%A7%92%E9%97%AE%E9%A2%98/)

```

---

* This string of code runs successfully but does not return an image

<img width="962" alt="image" src="https://github.com/JuliaLang/IJulia.jl/assets/62456287/237fceba-c7be-4899-9c6e-386518ad277c">

```julia
using Plots
Plots.plotly()
Plots.plot(Plots.fakedata(50, 5), w=3)
```

---

* Tried the `Base.istextmime(::MIME "application/vnd.plotly.v1+json") = true` that you mentioned, and the above code successfully returned the image

<img width="965" alt="image" src="https://github.com/JuliaLang/IJulia.jl/assets/62456287/c5df219b-1195-4249-b13d-9d3fe4eba51c">

```julia
Base.istextmime(::MIME"application/vnd.plotly.v1+json") = true

using Plots
Plots.plotly()
Plots.plot(Plots.fakedata(50, 5), w=3)
```

**Comment by stevengj** [comment]
> Tried the `Base.istextmime(::MIME "application/vnd.plotly.v1+json") = true` that you mentioned, and the above code successfully returned the image

i.e. you're having the problem I identified above, which is solved in the latest release (1.24.2).  Have you tried updating IJulia?

**Comment by a15355447898a** [comment]
> > Tried the `Base.istextmime(::MIME "application/vnd.plotly.v1+json") = true` that you mentioned, and the above code successfully returned the image
> 
> i.e. you're having the problem I identified above, which is solved in the latest release (1.24.2). Have you tried updating IJulia?

I have confirmed that I have upgraded IJulia to 1.24.2

![image](https://github.com/JuliaLang/IJulia.jl/assets/62456287/0f8319f3-8bb7-4c9a-bf0e-b7eae1d75b09)

**Comment by a15355447898a** [comment]
> > > Tried the `Base.istextmime(::MIME "application/vnd.plotly.v1+json") = true` that you mentioned, and the above code successfully returned the image
> > 
> > 
> > i.e. you're having the problem I identified above, which is solved in the latest release (1.24.2). Have you tried updating IJulia?
> 
> I have confirmed that I have upgraded IJulia to 1.24.2
> 
> ![image](https://user-images.githubusercontent.com/62456287/243153384-0f8319f3-8bb7-4c9a-bf0e-b7eae1d75b09.png)

Just suddenly found that the image in the jupyter notebook normal display out, but in the vscode jupyter will not display out

<img width="1189" alt="image" src="https://github.com/JuliaLang/IJulia.jl/assets/62456287/ceb0d1d1-419f-478b-86b3-220ccc181f75">

**Comment by stevengj** [comment]
> but in the vscode jupyter will not display out

Maybe you need to restart vscode?

**Comment by ajw-free** [comment]
stevengj - thanks for fixing this bug.  

 IJulia + Plots + plotly() backend now works again for me after updating Ijulia from v1.24.0 to v1.24.2.

**Comment by ajw-free** [comment]
This bug is currently present on Windows (tested 6 Sep 2023).  Plotly backend is not displaying a plot.

On a Windows 10 PC, I installed Julia 1.9.3 +  IJulia + Plots + plotly().

No output is shown within Jupyter notebook if I execute:

using Plots 
plotly()
display(plot(rand(10)))

I tested this on two Windows PCs;  finding the same problem on both.

Oddly, on my Ubuntu installation, running the same version of Julia 1.9.3 and updated packages, it correctly displays the plot.

**Comment by stevengj** [comment]
The old bug was fixed.  If this is a new bug, it should be a new issue.   It's hard to tell from your description what is now causing the problem for you.

What is the value of `keys(IJulia.display_dict(display(plot(rand(10)))))`?  This is what MIME types it is trying to display.

**Comment by ajw-free** [comment]
> What is the value of `keys(IJulia.display_dict(display(plot(rand(10)))))`? This is what MIME types it is trying to display.

On Windows 11 (which shows no plot display, just a blank space) I get:

KeySet for a Dict{String, Union{String, JSON.Writer.JSONText}} with 1 entry. Keys:
  "text/plain"

On my Ubuntu Linux, which does display the plot, the keys(...) output gives the same text as Windows: 
KeySet for a Dict{String, Union{String, JSON.Writer.JSONText}} with 1 entry. Keys:
  "text/plain"

**Comment by ajw-free** [comment]
I am not sure if this is a new bug, or whether the original issue (closed 4 June 2023) was not fully resolved for all platforms.
At the time of closure, the problem was fixed for me running Julia on Ubuntu Linux (and still is fixed).

I recently installed Julia on a Windows PC, and noticed that the issue was present on Windows, 
i.e. Within Jupyter Notebook using Plots, the plotly() backend does not display the plot.  
(with the same symptoms as I reported above on 12 May and 22 May 2023)

I do not know if the problem was ever solved for Windows installations, or whether it is a new issue on Windows.  Looking back on the bug reports above, nobody mentions Windows, so perhaps the bug was not ever solved for Windows.
